src/imgix-core-js.js

```diff
@@ -44,47 +38,25 @@
         this.settings[key] = val;
       }
 
-      if (Array.isArray(this.settings.domains)) { 
-        if (this.settings.domains.length > 1) {
-          console.warn("Warning: Domain sharding has been deprecated and will be removed in the next major version.\nAs a result, the 'domains' argument will be deprecated in favor of 'domain' instead.");
-        }
-        else if (this.settings.domains.length == 0) {
-          if (typeof(this.settings.domain) != "string" && this.settings.domain != null) {
-            throw new Error('ImgixClient.settings.domain only accepts a string argument');
-          }
-          else {
-            this.settings.domains = [this.settings.domain];
-          }
-        }
-      }
-      else {
-        this.settings.domains = [this.settings.domains];
+      if (typeof(this.settings.domain) != "string") {
+        throw new Error('ImgixClient.settings.domain only accepts a string argument');
       }
-
-      if (!this.settings.host && this.settings.domains == 0) {
-        throw new Error('ImgixClient must be passed valid domain(s)');
-      }
-
-      if (this.settings.shard_strategy !== SHARD_STRATEGY_CRC
-          && this.settings.shard_strategy !== SHARD_STRATEGY_CYCLE) {
-        throw new Error('Shard strategy must be one of ' +
-          SHARD_STRATEGY_CRC + ' or ' + SHARD_STRATEGY_CYCLE);
+      else if (!this.settings.host && this.settings.domain == '') {
+        throw new Error('ImgixClient must be passed valid domain');
       }
 
       if (this.settings.host) {
-        console.warn("'host' argument is deprecated; either use 'domain' or 'domains' instead.");
-        if (this.settings.domains.length == 0)
-          this.settings.domains[0] = this.settings.host;
+        console.warn("'host' argument is deprecated; use 'domain' instead.");
+        if (this.settings.domain.length == 0)
+          this.settings.domain = this.settings.host;
```

Comment:
Should we go ahead and remove `host` as an option too, since we're already bumping the major version here?

Reply:
Yea I was planning to do that in a separate PR